File: app/src/main/java/com/example/plantonic/ui/logInSignUp/login/LoginViewModel.java
package com.example.plantonic.ui.logInSignUp.login;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.airbnb.lottie.L;
import com.example.plantonic.repo.LoginRepository;

public class LoginViewModel extends ViewModel {
    private final LoginRepository loginRepository = new LoginRepository();

    public LiveData<Boolean> checkIfUserExists(String userId){
        loginRepository.checkIfUserExists(userId);
        return loginRepository.userExists;
    }
}

File: app/src/main/java/co/in/plantonic/ui/activity/logInSignUp/login/LoginViewModel.kt
package co.`in`.plantonic.ui.activity.logInSignUp.login

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import co.`in`.plantonic.firebaseClasses.UserItem
import co.`in`.plantonic.repo.LoginRepository

class LoginViewModel : ViewModel() {
    private val loginRepository = LoginRepository()

    fun checkIfUserExists(userId: String): LiveData<Boolean?> {
        loginRepository.checkIfUserExists(userId)
        return loginRepository.userExists
    }

    fun registerUser(userItem: UserItem): LiveData<Boolean?> {
        return loginRepository.registerNewUser(userItem)
    }

    fun getUserToken (uid : String) : LiveData<String?> {
        return loginRepository.getJwtToken(uid)
    }

}